src/transformers/trainer.py

```diff
@@ -611,33 +569,20 @@ def __init__(
                         f"FP16 provided in SM_HP_MP_PARAMETERS is {smp.state.cfg.fp16}, "
                         "but SageMaker Model Parallelism < 1.10 does not support FP16 in trainer."
                     )
-
-        if (args.fp16 or args.bf16) and self.sharded_ddp is not None:
+        if args.fp16 or args.bf16:
```

Comment:
Is this change OK to do? Previously we were only setting `args.half_precision_backend` if `self.sharded_ddp` was set (I'm guessing yes because of the removal of "cuda_amp" as an option)

@muellerzr You'll have a better idea of how this is managed :)

Reply:
I prefer keeping the current logic for now and opening another PR to handle logic related to the training argument `half_precision_backends` :-)  WDYT @muellerzr

Reply:
@statelesshz We can't merge in a PR that will possibly silently change behaviour and then bring in another PR later which rectifies. If this is done in two steps, then the second PR you propose would need to be added first.

Reply:
After further investigation, I think the current modification for mixed precision setup is acceptable.
FYI, https://github.com/huggingface/transformers/pull/23148 shifts `Trainer`'s mixed precision handling to accelerate. From https://github.com/huggingface/transformers/pull/23148#issue-1695779875 we know, `(fairscale)Sharded DDP and Apex are cases not supported in Accelerate `, so @pacman100 inserted `self.sharded_ddp is not None` at line 617 of `trainer.py`

And in this PR, we will remove fairscale ShardedDDP completely and the mixed precision logic needs to be handled synchronously.
Firstly I removed the `use_cuda_amp` related logic since Accelerate will handle it automatically. Secondly, we keep the logic of the Apex as Apex is not supported by Accelerate. Finally, I was going to remove use_cpu_amp but ended up keeping it